src/NuGet.Clients/NuGet.CommandLine/Commands/RestoreCommand.cs

```diff
@@ -818,7 +818,7 @@ private static bool IsSolutionOrProjectFile(string fileName)
         /// </summary>
         private static bool IsPackagesConfig(string projectFileName)
         {
-            return string.Equals(projectFileName, Constants.PackageReferenceFile)
+            return string.Equals(projectFileName, Constants.PackageReferenceFile, StringComparison.Ordinal)
```

Comment:
This method seems to be comparing paths. I suggest using the helper method to compare string paths based on OS.

```suggestion
            return string.Equals(projectFileName, Constants.PackageReferenceFile, PathUtil.GetStringComparisonBasedOnOS())
```